### ISSUE TITLE
Regarding the code implementation question

### ISSUE THREAD

**Issue by Yaoyongchen** [hoiliu-0801/DQ-DETR#18]
While reading the code, I have some doubts about the `num_select` obtained after the prediction in the CCM module. Is it taking the maximum predicted value among the batch samples as `num_select` to determine the number of queries for all batch samples in this input? Rather than selecting a different number of queries for different images? Is this choice made due to tensor computation constraints, or is there a misunderstanding on my part? I would appreciate it if you could take the time to clarify this for me. Thank you!

**Comment by Katie0723** [comment]
In the current code, when using multiple batches, the number of queries is set to the maximum value across all images in the batch for that epoch. This setup "just" ensures that multiple batches can run without causing errors since in our training process, we set the batch size to 1 due to GPU memory limitations.

When using multiple batches, it is still possible to apply some attention mask so that each sample in the batch can have a different number of queries.
